lib/cliapp/src/Obelisk/CliApp/Logging.hs

```diff
@@ -107,21 +125,13 @@ handleLog' noColor output = do
       hFlush stdout
   return $ isOverwrite output
 
--- | Log a message to the console.
---
--- Logs safely even if there are ongoing spinners.
-putLog :: Cli m => Severity -> Text -> m ()
-putLog sev = logMessage . Output_Log . WithSeverity sev
-
 -- | Like `putLog` but without the implicit newline added.
 putLogRaw :: Cli m => Severity -> Text -> m ()
 putLogRaw sev = logMessage . Output_LogRaw . WithSeverity sev
 
 -- | Like `putLog Alert` but also abrupts the program.
-failWith :: (MonadIO m, Cli m) => Text -> m a
-failWith s = do
-  putLog Alert s
-  liftIO $ exitWith $ ExitFailure 2
+failWith :: MonadError Text m => Text -> m a
+failWith = throwError
```

Comment:
This is missing a `putLog Alert`, no?

Reply:
nevermind - just noticed that you defined a `MonadError` instance that does it. cool.